os/adding-swap.md

```diff
@@ -0,0 +1,95 @@
+# Adding swap in CoreOS
+
+Swap is the process of moving pages of memory to a designated part of the hard disk, freeing up space when needed. Swap can be used to alleviate problems with low-memory environments.
+
+By default CoreOS does not include a partition for swap, however one can configure their system to have swap.
+
+## Managing swap with systemd
+
+Systemd can be used to create, begin, and stop a swapfile on your system. Write the following systemd unit file and enable + start the service to add swap to your instance of CoreOS.
+
+### Creating the systemd unit file
+
+The following systemd unit file creates, starts, and stops a 1GiB swapfile. It should be written to `/etc/systemd/system/swap.service`.
+
+```
+[Unit]
+Description=Turn on swap
+
+[Service]
+Type=oneshot
+Environment="SWAP_PATH=/var/vm" "SWAP_FILE=1GiB.swap"
+ExecStartPre=-/usr/bin/rm -rf ${SWAP_PATH}
+ExecStartPre=/usr/bin/mkdir -p ${SWAP_PATH}
+ExecStartPre=/usr/bin/touch ${SWAP_PATH}/${SWAP_FILE}
+ExecStartPre=/bin/bash -c "fallocate -l 1024m ${SWAP_PATH}/${SWAP_FILE}"
+ExecStartPre=/usr/bin/chmod 600 ${SWAP_PATH}/${SWAP_FILE}
+ExecStartPre=/usr/sbin/mkswap ${SWAP_PATH}/${SWAP_FILE}
+ExecStartPre=/usr/sbin/sysctl vm.swappiness=10
+ExecStart=/sbin/swapon ${SWAP_PATH}/${SWAP_FILE}
+ExecStop=/sbin/swapoff ${SWAP_PATH}/${SWAP_FILE}
+ExecStopPost=-/usr/bin/rm -rf ${SWAP_PATH}
+RemainAfterExit=true
+
+[Install]
+WantedBy=multi-user.target
+```
+
+**Note** Ensure the block device containing your swapfile is configured to auto-mount on boot.
+
+### Enable the unit and start using swap
+
+The following command enables and starts the new `swap` service.
+
+```
+$ systemctl enable --now /etc/systemd/system/swap.service
+```
+
+Swap has been enabled and will be started automatically on subsequent reboots. We can verify that the swap is being used by running `free -hm`:
+
+```
+$ free -hm
+             total       used       free     shared    buffers     cached
+[...]
+Swap:         1.0G         0B       1.0G
+```
+
+## Problems and Considerations
+
+### Btrfs and XFS
+
+Due to performance problems, it is not advised that one creates a swapfile on btrfs. If your system uses btrfs, you should create a designated partition for swap.
+
+### Partition size
+
+The partition on which you are creating the swapfile must be large enough for your swapfile size.
+
+### Verifying that your system can use a swapfile
+
+To verify that our system meets those requirements, run the following command:
+
+```
+$ df -Th
+Filesystem     Type      Size  Used Avail Use% Mounted on
+[...]
+/dev/sdXN      ext4      2.0G  3.0M  1.8G   1% /var
+```
+
+The block device mounted at `/var/` (`/dev/sdXN`) is the correct filesystem type and has enough space for a 1GiB swapfile, so everything should work.
+
+## Adding swap with ignition
+
+The following ignition file will configure a 1GiB swapfile located at `/var/vm/1GiB.swap`.
```

Comment:
the Ignition excerpt below creates `/var/vm/swapfile1`, actually. Make the above unit file match this -- "swapfile1" -- rather than putting the nominal size in the file name. Make this sentence match them both.